system/datatypes/peer_reviewed_article.php

```diff
@@ -437,6 +440,32 @@ public function editAction(HyphaRequest $request) {
 
 		$this->xml->lockAndReload();
 
+		$newImage = $form->dataFor(self::FIELD_NAME_FEATURED_IMAGE);
+		if ($_FILES[self::FIELD_NAME_FEATURED_IMAGE]) {
+			$ext = strtolower(substr(strrchr($_FILES[self::FIELD_NAME_FEATURED_IMAGE]['name'], '.'), 1));
+			@$size = getimagesize($_FILES[self::FIELD_NAME_FEATURED_IMAGE]['tmp_name']);
+			if ($size && in_array($ext, array('jpg','jpeg','png','gif','bmp'))) {
+				$maxSize = [1120, 800];
+				$needResize = $size[0] > $maxSize[0] || $size[1] > $maxSize[1];
+				$filename = uniqid() . '.' . $ext;
+				$destinations = ['org' => 'data/images/org/' . $filename, 'img' => 'data/images/' . $filename];
+				if ($needResize && !file_exists('data/images/org/')) {
+					mkdir('data/images/org/', 0777, true);
+				}
+				$orgUrlIndex = $needResize ? 'org' : 'img';
+				if (move_uploaded_file($_FILES[self::FIELD_NAME_FEATURED_IMAGE]['tmp_name'], $destinations[$orgUrlIndex])) {
+					if ($needResize) {
+						image_resize($destinations[$orgUrlIndex], $destinations['img'], $maxSize[0], $maxSize[1]);
+					}
+				}
+
+				$newImage = $destinations['img'];
+			}
```

Comment:
This silently ignores images that are too big or have an invalid extension.

Reply:
fixed